pkg/services/accesscontrol/resourceservices/resource_services.go

```diff
@@ -0,0 +1,93 @@
+package resourceservices
```

Comment:
Would we run into any problems if we made this thing just for teams and placed it somewhere like in `pkg/services/teams` instead?

Reply:
The main reason for this placement was that we've put the DS resource service under `accesscontrol` - https://github.com/grafana/grafana-enterprise/blob/main/src/pkg/extensions/accesscontrol/resource_services.go.

Also, we don't currently have a `pkg/services/teams` - the bulk of team backend code is under `pkg/api`.

That said, I don't think it should cause any problems. So, if you think that placement would make more sense, I can move it.

Reply:
> The main reason for this placement was that we've put the DS resource service under accesscontrol - https://github.com/grafana/grafana-enterprise/blob/main/src/pkg/extensions/accesscontrol/resource_services.go.

🤔 Shouldn't teams just be added to that one, if we want to go with the resource service approach?

> Also, we don't currently have a pkg/services/teams - the bulk of team backend code is under pkg/api.
>
> That said, I don't think it should cause any problems. So, if you think that placing this under pkg/services/teams would make more sense, I can move it.

I think that would create a nicer to work with API, but it's not urgent or important enough to block anything. Internal APIs can always be shuffled around when we need to 🙂

Reply:
> 🤔 Shouldn't teams just be added to that one, if we want to go with the resource service approach?

It's in enterprise code-base, but we need to use teams resource service in OSS code, so that wouldn't work unless we move the DS resource service to OSS.

I quite like the idea of having a `pkg/services/teams` - I'll keep it out of these PRs, as there's already a lot going on, but will add an issue to refactor the structure afterwards.

Reply:
Ahhh.

Hah. It's good someone has their brain with them even in Januaries